.github/workflows/lint.yaml

```diff
@@ -0,0 +1,57 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Golang with cache
+        uses: magnetikonline/action-golang-cache@v4
+        with:
+          go-version-file: go.mod
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.64.6
+          working-directory: ${{ inputs.cwd }}
+
+      - name: staticcheck
+        uses: dominikh/staticcheck-action@v1
+        with:
+          version: 2025.1.1
+          install-go: false
+          working-directory: ${{ inputs.cwd }}
+          cache-key: ${{ inputs.cwd }}
```

Comment:
why not run `staticcheck` as part of golangci-lint? I know we pulled it out in another project because it was causing OOMs (largely due to the very large API surface of a dependent package), but if it can run on dev machines without requiring the installation of another program it may be worth the switch

Reply:
No particular reason.  This is copy-pasted from another functioning environment we have running.  And I dont want to futz with it if that isn't required at the moment.